go-controller/pkg/ovn/service.go

```diff
@@ -43,38 +43,33 @@ func (ovn *Controller) syncServices(services []interface{}) {
 			continue
 		}
 
-		if !util.IsClusterIPSet(service) {
-			klog.V(5).Infof("Skipping service %s due to clusterIP = %q",
-				service.Name, service.Spec.ClusterIP)
-			continue
-		}
-
 		for _, svcPort := range service.Spec.Ports {
-			protocol, err := util.ValidateProtocol(svcPort.Protocol)
-			if err != nil {
+			if err := util.ValidatePort(svcPort.Protocol, svcPort.Port); err != nil {
 				klog.Errorf("Error validating protocol for port %s: %v", svcPort.Name, err)
 				continue
 			}
 
-			if util.ServiceTypeHasNodePort(service) {
-				port := fmt.Sprintf("%d", svcPort.NodePort)
-				nodeportServices[protocol] = append(nodeportServices[protocol], port)
-			}
-
-			if svcPort.Port == 0 {
-				continue
-			}
-
-			key := util.JoinHostPortInt32(service.Spec.ClusterIP, svcPort.Port)
-			clusterServices[protocol] = append(clusterServices[protocol], key)
-
 			if len(service.Spec.ExternalIPs) == 0 {
 				continue
 			}
 			for _, extIP := range service.Spec.ExternalIPs {
 				key := util.JoinHostPortInt32(extIP, svcPort.Port)
-				lbServices[protocol] = append(lbServices[protocol], key)
+				lbServices[svcPort.Protocol] = append(lbServices[svcPort.Protocol], key)
+
+			}
+
+			if !util.IsClusterIPSet(service) {
+				klog.V(5).Infof("Skipping service %s due to clusterIP = %q", service.Name, service.Spec.ClusterIP)
+				break
```

Comment:
did you mean continue?

Reply:
Will rectify with the previous comment you gave me on line 46